merge.go

```diff
@@ -484,16 +484,18 @@ func mergeTermFreqNormLocsByCopying(term []byte, postItr *PostingsIterator,
 	lastDocNum uint64, lastFreq uint64, lastNorm uint64, err error) {
 	nextDocNum, nextFreq, nextNorm, nextFreqNormBytes, nextLocBytes, err :=
 		postItr.nextBytes()
-	for err == nil && len(nextFreqNormBytes) > 0 {
+	for err == nil {
```

Comment:
Not sure I follow this entirely - why change this?

Reply:
Now that we can have optional freqNorm bytes independent of locationBytes, we can't skip the merging/copying of the location bytes even if there are no freqNorm bytes configured/present on the disk.

Until now, zap always had some non-nil `nextFreqNormBytes` whenever there are any valid location bytes present.  ie location data was optional but freqNorm data wasn't.
Due to this - many of the code flow has that assumption while making disk-level iterator advancements. This is one such case.

Reply:
Now that we can have optional freqNorm bytes independent of locationBytes, we can't skip the merging/copying of the location bytes even if there are no freqNorm bytes configured/present on the disk.

Until now, zap always had some non-nil nextFreqNormBytes whenever there are any valid location bytes present. ie location data was optional but freqNorm data wasn't.
Due to this - many of the code flow has that assumption while making disk-level iterator advancements. This is one such case.